src/modules/person/domain/person-time-limit-info.service.ts

```diff
@@ -41,6 +45,36 @@ export default class PersonTimeLimitService {
             lockInfos.push(new PersonTimeLimitInfo(TimeLimitOccasion.NO_KONTEXTE, noKontexteDeadline));
         }
 
-        return lockInfos;
+        const personenKontexte: Personenkontext<true>[] = await this.dBiamPersonenkontextRepo.findByPerson(personId);
+
+        const timeLimitInfosPromises: Promise<PersonTimeLimitInfo | null>[] = personenKontexte.map(
+            async (personenKontext: Personenkontext<true>) => {
+                if (personenKontext.befristung != null) {
+                    const personenKontextExpires: Date = new Date(personenKontext.befristung);
+
+                    const currentDate: Date = new Date();
+                    const timeDiffMs: number = personenKontextExpires.getTime() - currentDate.getTime();
+                    const timeDiffDays: number = Math.floor(timeDiffMs / (1000 * 3600 * 24));
+                    if (timeDiffDays <= KONTEXT_EXPIRES_IN_DAYS) {
+                        const organisation: Option<Organisation<true>> = await personenKontext.getOrganisation();
+                        const rolle: Option<Rolle<true>> = await personenKontext.getRolle();
+
+                        return new PersonTimeLimitInfo(
+                            TimeLimitOccasion.PERSONENKONTEXT_EXPIRES,
+                            personenKontextExpires,
+                            organisation?.name,
+                            rolle?.name,
+                        );
+                    }
+                }
+                return null;
+            },
+        );
+
+        const timeLimitInfos: (PersonTimeLimitInfo | null)[] = await Promise.all(timeLimitInfosPromises);
+        const validTimeLimitInfos: PersonTimeLimitInfo[] = timeLimitInfos.filter(
+            (info: PersonTimeLimitInfo | null): info is PersonTimeLimitInfo => info !== null,
+        );
```

Comment:
Less of a problem here because we can be sure `info` is not `undefined`, but I'd still prefer this
```suggestion
        const validTimeLimitInfos: PersonTimeLimitInfo[] = timeLimitInfos.filter(
            (info: PersonTimeLimitInfo | null): info is PersonTimeLimitInfo => !!info,
        );
```